karabo_bridge/simulation.py

```diff
@@ -78,29 +94,27 @@ def gen_combined_detector_data(source, tid_counter):
     gen[source]['image.trainId'] = trainId
     gen[source]['image.status'] = status
 
-    checksum = np.ones(16, dtype=np.int8)
-    magicNumberEnd = np.ones(8, dtype=np.int8)
+    checksum = bytes(np.ones(16, dtype=np.int8))
+    magicNumberEnd = bytes(np.ones(8, dtype=np.int8))
     status = 0
     trainId = tid_counter
 
     gen[source]['trailer.checksum'] = checksum
     gen[source]['trailer.magicNumberEnd'] = magicNumberEnd
     gen[source]['trailer.status'] = status
-    gen[source]['trailer.trainId'] = trainId
 
-    data = np.ones(416, dtype=np.uint8)
+    data = bytes(np.ones(416, dtype=np.uint8))
```

Comment:
Why are we converting all these arrays to bytes? Won't this break the `isinstance(value, np.ndarray)` check below?

Reply:
It looks like it makes it right for protocol version 1.0 - I think I recall that's the format that the real system uses at present - but at the cost of breaking protocol version 2.1, because the arrays will never be extracted and sent separately. The simulation server is meant to support both protocol versions, I think.

Reply:
clarified offline.